app/src/main/java/com/google/wear/whereami/data/LocationViewModel.kt

```diff
@@ -14,33 +14,63 @@
 package com.google.wear.whereami.data
 
 import android.Manifest
-import android.content.ComponentName
 import android.content.Context
 import android.content.pm.PackageManager
-import androidx.wear.complications.datasource.ComplicationDataSourceUpdateRequester
+import com.dropbox.android.external.store4.Fetcher
+import com.dropbox.android.external.store4.MemoryPolicy
+import com.dropbox.android.external.store4.Store
+import com.dropbox.android.external.store4.StoreBuilder
 import com.google.android.gms.location.LocationRequest
 import com.patloew.colocation.CoGeocoder
 import com.patloew.colocation.CoLocation
+import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.FlowPreview
 import kotlinx.coroutines.withContext
+import java.io.IOException
+import kotlin.time.Duration
 
-class LocationViewModel(val applicationContext: Context) {
+class LocationViewModel(private val applicationContext: Context, private val coroutineScope: CoroutineScope) {
     private val coGeocoder = CoGeocoder.from(applicationContext)
     private val coLocation = CoLocation.from(applicationContext)
 
-    suspend fun readLocationResult(): LocationResult {
+    @OptIn(FlowPreview::class, kotlinx.coroutines.ExperimentalCoroutinesApi::class,
+        kotlin.time.ExperimentalTime::class
+    )
+    val store: Store<String, LocationResult> = StoreBuilder
+        .from<String, LocationResult>(Fetcher.of { readLocationResult() })
+        .cachePolicy(MemoryPolicy.builder<String, LocationResult>()
+            .setMaxSize(10)
+            .setExpireAfterWrite(Duration.Companion.minutes(1))
+            .build())
+        .build()
+
+    private suspend fun readLocationResult(): LocationResult {
```

Comment:
One optional suggestion: you could keep this public, and make the store private. That way, the caching mechanism is encapsulated and somewhat hidden from consumers. (i.e we don't want them messing with other parts of the store, clearing it, etc.)

That way you wouldn't have to change a lot of the other files either, since they'd be using the same API... (and maybe you'd add an optional/default parameter for skipping the cache).

Reply:
I like that will do that as part of removing dropbox store.